hrana/test/hrana_test.dart

```diff
@@ -10,107 +10,113 @@ void main() {
 
   late Database database;
 
-  setUpAll(() async {
-    port = await selectFreePort();
-    server = await startSqld(port);
-  });
-
-  tearDownAll(() async {
-    await server.stop();
-  });
-
-  setUp(() async {
-    database = await Database.connect(Uri.parse('ws://localhost:$port/'));
-
-    await database
-        .execute('CREATE TABLE users (name TEXT, CHECK (LENGTH(name) < 10));');
-  });
-
-  tearDown(() async {
-    await database.execute('DROP TABLE users;');
-
-    await database.close();
-  });
-
-  test('can select statements', () async {
-    final result = await database.select('SELECT 1 AS r;');
-    expect(result.columnNames, ['r']);
-    expect(result.rows, [
-      [1]
-    ]);
-  });
-
-  test('can execute statements', () async {
-    final result = await database.execute(
-      'INSERT INTO users (name) VALUES (?);',
-      arguments: ['name'],
-    );
-    expect(result.affectedRows, 1);
-    expect(result.lastInsertRowId, 1, skip: 'rowid appears to not be sent');
-  });
-
-  group('stored statements', () {
-    test('can be executed', () async {
-      final stored =
-          await database.storeSql('INSERT INTO users (name) VALUES (?);');
-      final result = await stored.execute(arguments: ['Stored']);
-      expect(result.affectedRows, 1);
-    });
+  for (final scheme in const ['http', 'ws']) {
+    group(scheme, () {
+      setUpAll(() async {
+        port = await selectFreePort();
+        server = await startSqld(port);
+      });
 
-    test('can be selected', () async {
-      final stored = await database.storeSql('SELECT 1');
-      final result = await stored.select();
-      expect(result.rows, [
-        [1]
-      ]);
-    });
+      tearDownAll(() async {
+        await server.stop();
+      });
```

Comment:
I think we can move this out of the `group` since the server supports both protocols right? Doesn't change much but might make the tests run slightly faster.

Reply:
Yep, good catch 👍